### PR TITLE
Switch to create emails using buffalo templates

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

File: application/listeners/posthelpers.go [hunk 1]
```diff
@@ -257,6 +265,8 @@ func join(s1, s2 string) string {
 	return s1 + "-" + s2
 }
 
+var getT = notifications.GetEmailTemplate
```
Comment (Schparky) [inline application/listeners/posthelpers.go]: stylistic nit: I'd prefer to see this near the point of use or at the top of the file.

File: application/listeners/posthelpers_test.go [hunk 2]
```diff
@@ -112,23 +112,27 @@ func (ms *ModelSuite) TestSendNotificationRequestFromStatus() {
 		domain.ErrLogger.SetOutput(os.Stderr)
 	}()
 
+	var getT = notifications.GetEmailTemplate
```
Comment (Schparky) [inline application/listeners/posthelpers_test.go]: Depending on whether you change the scope of the other `getT`, you could combine them into one.

File: application/notifications/email.go [hunk 3]
```diff
@@ -1,5 +1,47 @@
...
+var emailTemplates = map[string]string{
+	domain.MessageTemplateNewThreadMessage:                domain.MessageTemplateNewThreadMessage,
+	domain.MessageTemplateNewRequest:                      domain.MessageTemplateNewRequest,
+	domain.MessageTemplateRequestFromCommittedToOpen:      domain.MessageTemplateRequestFromCommittedToOpen,
+	domain.MessageTemplateRequestFromAcceptedToOpen:       domain.MessageTemplateRequestFromAcceptedToOpen,
+	domain.MessageTemplateRequestFromOpenToCommitted:      domain.MessageTemplateRequestFromOpenToCommitted,
+	domain.MessageTemplateRequestFromCommittedToAccepted:  domain.MessageTemplateRequestFromCommittedToAccepted,
+	domain.MessageTemplateRequestFromDeliveredToAccepted:  domain.MessageTemplateRequestFromDeliveredToAccepted,
+	domain.MessageTemplateRequestFromReceivedToAccepted:   domain.MessageTemplateRequestFromReceivedToAccepted,
+	domain.MessageTemplateRequestFromCommittedToDelivered: "request_delivered",
```
Comment (Schparky) [inline application/notifications/email.go]: Do you want a `const` for this?

File: application/notifications/sendgrid.go [hunk 4]
```diff
@@ -42,29 +43,23 @@ func (e *SendGridService) Send(msg Message) error {
...
+	bodyBuf := &bytes.Buffer{}
+	if err := r.HTML(msg.Template).Render(bodyBuf, msg.Data); err != nil {
+		return errors.New("error rendering message body - " + err.Error())
+	}
+	body := bodyBuf.String()
 
-	request := sendgrid.GetRequest(apiKey, SendGridEndpointMailSend, SendGridAPIUrl)
-	request.Method = "POST"
-	request.Body = mail.GetRequestBody(message)
+	m := mail.NewSingleEmail(from, subject, to, body, body)
```
Comment (Schparky) [inline application/notifications/sendgrid.go]: Are you stripping HTML tags for the plain-text version? I have some code in a stash that does this if you're interested.

File: application/listeners/posthelpers.go (hunk 1 above)
Comment (Baggerone) [reply application/listeners/posthelpers.go]: I must have been calling it more than once in previous iteration before I finalized the code here.

File: application/notifications/email.go (hunk 3 above)
Comment (Baggerone) [reply application/notifications/email.go]: Yep

**Pull request merged**